tests/models/virtualfolder.py

```diff
@@ -152,3 +152,118 @@ def test_vfolder_with_no_filter_rules():
         vfolder.clean_fields()
 
     assert u'Some filtering rule must be specified.' in str(excinfo.value)
+
+
+@pytest.mark.django_db
+def test_vfolder_membership(site_matrix):
+    from pootle_pytest.factories import VirtualFolderFactory
+
+    from pootle_store.models import Unit
+    from pootle_store.util import OBSOLETE, TRANSLATED
+
+    vfolder = VirtualFolderFactory(filter_rules="store0.po")
+
+    live_units = Unit.objects.filter(state__gt=OBSOLETE)
+
+    expected_units = live_units.filter(store__name="store0.po")
+
+    # check default vfolder membership
+    assert (
+        sorted(vfolder.units.values_list("pk", flat=True))
+        == sorted(expected_units.values_list("pk", flat=True)))
+
+    vfolder.location = "/language0/{PROJ}/"
+    vfolder.save()
+
+    expected_units = live_units.filter(
+        store__translation_project__language__code="language0",
+        store__name="store0.po")
+
+    # check vfolder membership after changing the location
+    assert (
+        sorted(vfolder.units.values_list("pk", flat=True))
+        == sorted(expected_units.values_list("pk", flat=True)))
+
+    obsolete_unit = (
+        Unit.objects.filter(
+            state=OBSOLETE,
+            store__translation_project__language__code="language0",
+            store__name="store0.po"))[0]
+
+    # obsolete unit is not in the vfolder
+    assert obsolete_unit not in vfolder.units.all()
+
+    obsolete_unit.state = TRANSLATED
+    obsolete_unit.save()
+
+    # unobsoleted unit is in the vfolder
+    assert obsolete_unit in vfolder.units.all()
+
+    to_obsolete = vfolder.units.all()[0]
+    to_obsolete.state = OBSOLETE
+    to_obsolete.save()
+
+    # obsoleted unit is not in the vfolder
+    assert to_obsolete not in vfolder.units.all()
+
+
+@pytest.mark.django_db
+def test_vfolder_unit_priorities(site_matrix):
+    from pootle_pytest.factories import VirtualFolderFactory
+
+    from pootle_store.models import Unit
+
+    assert all(
+        priority == 1
+        for priority
+        in Unit.objects.values_list("priority", flat=True))
+
+    vfolder0 = VirtualFolderFactory(filter_rules="store0.po", priority=3)
+
+    assert all(
+        priority == 3
+        for priority
+        in vfolder0.units.values_list("priority", flat=True))
+
+    assert all(
+        priority == 1.0
+        for priority
+        in Unit.objects.filter(vfolders__isnull=True)
```

Comment:
Again assuming that there is no other virtual folder, thus you can be skipping some other stores besides `store0.po`. Perhaps it is better to check for all units that don't belong to that store rather than units that are not on any virtual folder.